DRIQuery.cpp

```diff
@@ -37,6 +40,20 @@ std::list<DriverConfiguration> DRIQuery::queryDriverConfigurationOptions(const G
         DriverConfiguration config;
         config.setScreen(i);
 
+
+        char *p = (char *)(*(this->getGlExtensionsString))(GL_EXTENSIONS);
+        if (p == nullptr) std::cout << "No GL extensions available" << std::endl;
```

Comment:
You are doing this same check on line 52. Why are you doing it two times?

Reply:
I wanted to check whether the driver supports functions "glXGetScreenDriver", "glXGetDriverConfig". Actually these should be checked in L53 only since these are gl**x**extensions but by any other way if there is a possibility to get these via any **gl** extension. I'm just checking these in string returned by **gl**GetString. It is different from L52 one.

Do you know in which glx extension these functions exists? So that I can check for that in possibleExts string.

Reply:
Well, for a driver to work under linux, it must support GLX, so I don't think you will need to do the other check.
Just checking it on L53 is enought. I don't see how checking this on lines 44 and 45 improve anything.